src/internal/builtInServices.ts

```diff
@@ -0,0 +1,175 @@
+                } else {
+                    return success([].concat.apply([], req.args));
+                }
+            },
+
+            string_to_b58: (req) => {
+                if (req.args.length !== 1) {
+                    return error('string_to_b58 accepts only one string argument');
+                } else {
+                    return success(encode(new TextEncoder().encode(req.args[0])));
+                }
+            },
+
+            string_from_b58: (req) => {
+                if (req.args.length !== 1) {
+                    return error('string_from_b58 accepts only one string argument');
+                } else {
+                    return success(new TextDecoder().decode(decode(req.args[0])));
+                }
+            },
+
+            bytes_to_b58: (req) => {
+                if (req.args.length !== 1 || !Array.isArray(req.args[0])) {
+                    return error('bytes_to_b58 accepts only single argument: array of numbers');
+                } else {
+                    const argumentArray = req.args[0] as number[];
+                    return success(encode(new Uint8Array(argumentArray)));
+                }
+            },
+
+            bytes_from_b58: (req) => {
+                if (req.args.length !== 1) {
+                    return error('bytes_from_b58 accepts only one string argument');
+                } else {
+                    return success(Array.from(decode(req.args[0])));
+                }
+            },
+        },
+
+        peer: {
+            timeout: (req) => {
+                if (req.args.length !== 2) {
+                    return error('timeout accepts exactly two arguments: timeout duration in ms and a message string');
+                }
+                const durationMs = req.args[0];
+                const message = req.args[1];
+
+                return new Promise((resolve) => {
+                    setTimeout(() => {
+                        const res = success(message);
+                        resolve(res);
+                    }, durationMs);
+                });
+            },
+
+            identify: (req) => {
+                return error('The JS implementation of Peer does not support identify');
+            },
+        },
+
+        security: {
+            sign: async (req) => {
+                if (req.args.length !== 1) {
+                    return error('sign accepts exactly one argument: data be signed in format of u8 array of bytes');
+                }
+
+                if (req.particleContext.initPeerId !== context.peerId) {
+                    return error('sign is only allowed to be called on the same peer the particle was initiated from');
+                }
+
+                const t = req.tetraplets[0][0];
+                const serviceFnPair = `${t.service_id}.${t.function_name}`;
+
+                const allowedServices = [
+                    'TrustGraph.get_trust_bytes',
+                    'TrustGraph.get_revocation_bytes',
```

Comment:
it's `trust-graph` 

https://github.com/fluencelabs/trust-graph/blob/master/aqua/trust-graph.aqua#L72